test/merger/test_merger.py

```diff
@@ -259,6 +259,110 @@ def test_no_merge_across_reset() -> None:
     ],
     ids=["anonymous_gate", "circuit_with_multi_and_single_qubit_gates", "unpacking_usecase", "circuit_with_4_qubits"],
 )
-def test_barriers(circuit: Circuit, expected_result: str) -> None:
+def test_merge_barriers(circuit: Circuit, expected_result: str) -> None:
+    from opensquirrel.merger.general_merger import merge_barriers
+
+    circuit.merge_single_qubit_gates()
+    circuit.ir.statements = merge_barriers(circuit.ir.statements)
+    assert str(circuit) == expected_result
+
+
+@pytest.mark.parametrize(
+    ("circuit", "expected_result"),
+    [
+        (
+            (
+                CircuitBuilder(2)
+                .H(1)
+                .Rz(1, Float(2.332))
+                .H(0)
+                .Rx(0, Float(1.423))
+                .barrier(0)
+                .barrier(1)
+                .H(1)
+                .Rz(1, Float(2.332))
+                .H(0)
+                .Rx(1, Float(1.423))
+                .barrier(0)
+                .barrier(1)
+                .to_circuit()
+            ),
+            """version 3.0
+
+qubit[2] q
+
+Anonymous gate: BlochSphereRotation(Qubit[0], axis=[ 0.60376 -0.52053  0.60376], angle=-2.18173, phase=1.5708)
+Anonymous gate: BlochSphereRotation(Qubit[1], axis=[0.36644 0.85525 0.36644], angle=-1.72653, phase=1.5708)
+barrier q[0]
+barrier q[1]
+H q[0]
+barrier q[0]
+Anonymous gate: BlochSphereRotation(Qubit[1], axis=[ 0.28903 -0.42028 -0.86013], angle=1.66208, phase=1.5708)
+barrier q[0]
+barrier q[1]
+""",
+        ),
+        (
+            CircuitBuilder(3)
+            .H(0)
+            .Ry(0, Float(2))
+            .barrier(2)
+            .barrier(0)
+            .barrier(1)
+            .H(0)
+            .H(1)
+            .Ry(1, Float(2))
+            .barrier(1)
+            .H(2)
+            .barrier(2)
+            .barrier(0)
+            .barrier(1)
+            .to_circuit(),
+            """version 3.0
+
+qubit[3] q
+
+Anonymous gate: BlochSphereRotation(Qubit[0], axis=[ 0.97706  0.      -0.21296], angle=3.14159, phase=1.5708)
+barrier q[2]
+barrier q[0]
+barrier q[1]
+Anonymous gate: BlochSphereRotation(Qubit[1], axis=[ 0.97706  0.      -0.21296], angle=3.14159, phase=1.5708)
+barrier q[1]
+H q[2]
+H q[0]
+barrier q[1]
+""",
+        ),
+        (
+            CircuitBuilder(2)
+            .H(0)
+            .Ry(0, Float(2))
+            .barrier(1)
+            .barrier(0)
+            .barrier(1)
```

Comment:
I don't know, according to the specifications laid out by Richard and Chris, it is totally viable to place two barrier directives after each other for the same qubit. This is what the front end application allows you to do,

Reply:
It would be good to have access to those specifications.

However, what I say here is that if:

```
barrier q[1]
barrier q[0]
barrier q[1]
```

should be optimized to:

```
barrier q[0]
barrier q[1]
```

(the order of these last two instructions shouldn't matter).

Reply:
Is it okay to ticket this? It would be good to add as a `Scheduler` optimization.

Reply:
Just a comment. If we ever do anything, we'll do it in a Scheduler pass, indeed.